packages/fxa-auth-server/fxa-oauth-server/test/api.js

```diff
@@ -942,6 +953,45 @@ describe('/v1', function() {
           assert(res.result.auth_at);
         });
 
+        it('returns an implicit JWT formatted token with sub rotation', async function() {
+          mockAssertion().reply(200, VERIFY_GOOD);
+          const notRotatedResult = await Server.api.post({
+            url: '/authorization',
+            payload: authParams({
+              client_id: ppidClient.id,
+              response_type: 'token',
+            }),
+          });
+
+          assert.equal(notRotatedResult.statusCode, 200);
+          assertSecurityHeaders(notRotatedResult);
+          assert(notRotatedResult.result.access_token);
+          assert.isNull(
+            validators.jwt.validate(notRotatedResult.result.access_token).error
+          );
+
+          mockAssertion().reply(200, VERIFY_GOOD);
+          const rotatedResult = await Server.api.post({
+            url: '/authorization',
+            payload: authParams({
+              client_id: ppidClient.id,
+              ppid_seed: 100,
+              response_type: 'token',
+            }),
+          });
+
+          assert.equal(rotatedResult.statusCode, 200);
+          assert(rotatedResult.result.access_token);
+          assert.isNull(
+            validators.jwt.validate(rotatedResult.result.access_token).error
+          );
+
+          const notRotatedJWT = decodeJWT(notRotatedResult.result.access_token);
+          const rotatedJWT = decodeJWT(rotatedResult.result.access_token);
+
+          assert.notEqual(notRotatedJWT.claims.sub, rotatedJWT.claims.sub);
```

Comment:
I would consider checking the known value of `notRotatedJWT.claims.sub` here, since you know it. (Otherwise maybe you got one of your booleans the wrong way round and it's accidentally `notRotatedJWT` that we rotated)

Reply:
I wasn't able to assert what you suggested here because PPIDs were enabled, meaning my test had some terrible names. Because of that, I reworked the test and improved testing did point out I had a bug - I was feeding the wrong length to hkdf. 👍